x-pack/test/fleet_api_integration/apis/policy_secrets.ts

```diff
@@ -267,5 +285,68 @@ export default function (providerContext: FtrProviderContext) {
 
       expectCompiledPolicyVars(agentPolicy);
     });
+
+    it('should allow secret values to be updated (single policy update API)', async () => {
+      const updatedPolicy = createdPolicyToUpdatePolicy(createdPackagePolicy);
+      updatedPolicy.vars.package_var_secret.value = 'new_package_secret_val';
+      const updateRes = await supertest
+        .put(`/api/fleet/package_policies/${createdPackagePolicyId}`)
+        .set('kbn-xsrf', 'xxxx')
+        .send(updatedPolicy)
+        .expect(200);
+
+      const updatedPackagePolicy = updateRes.body.item;
+
+      updatedPackageVarId = updatedPackagePolicy.vars.package_var_secret.value.id;
+      expect(updatedPackageVarId).to.be.an('string');
+      expect(
+        arrayIdsEqual(updatedPackagePolicy.secret_references, [
+          { id: updatedPackageVarId },
+          { id: streamVarId },
+          { id: inputVarId },
+        ])
+      ).to.eql(true);
+      expect(updatedPackagePolicy.inputs[0].streams[0].compiled_stream).to.eql({
+        'config.version': 2,
+        package_var_secret: secretVar(updatedPackageVarId),
+        input_var_secret: secretVar(inputVarId),
+        stream_var_secret: secretVar(streamVarId),
+      });
+      expect(updatedPackagePolicy.inputs[0].compiled_input).to.eql({
+        package_var_secret: secretVar(updatedPackageVarId),
+        input_var_secret: secretVar(inputVarId),
+      });
+      expect(updatedPackagePolicy.vars.package_var_secret.value.isSecretRef).to.eql(true);
+      expect(updatedPackagePolicy.vars.package_var_secret.value.id).eql(updatedPackageVarId);
+      expect(updatedPackagePolicy.inputs[0].vars.input_var_secret.value.isSecretRef).to.eql(true);
+      expect(updatedPackagePolicy.inputs[0].vars.input_var_secret.value.id).eql(inputVarId);
+      expect(
+        updatedPackagePolicy.inputs[0].streams[0].vars.stream_var_secret.value.isSecretRef
+      ).to.eql(true);
+      expect(updatedPackagePolicy.inputs[0].streams[0].vars.stream_var_secret.value.id).eql(
+        streamVarId
+      );
+    });
+
+    it('should have correctly deleted the secrets', async () => {
```

Comment:
It does depend on the result of the previous, I think I chose to do this because the previous test already has loads of assertions, and this test requires an ES call and further assertions.